src/library/scala/collection/mutable/ArrayBuilder.scala

```diff
@@ -46,7 +46,7 @@ sealed abstract class ArrayBuilder[T]
   protected[this] def resize(size: Int): Unit
 
   /** Add all elements of an array. */
-  def addAll(xs: Array[_ <: T]): this.type = doAddAll(xs, 0, xs.length)
+  def addAll(xs: Array[_ <: T]): this.type = addAll(xs, 0, xs.length)
```

Comment:
We can just forwarding when xs is not empty

Reply:
That is a micro-optimization too far.